util/journaldb/src/overlayrecentdb.rs

```diff
@@ -253,7 +253,7 @@ impl JournalDB for OverlayRecentDB {
 		let mut mem = self.transaction_overlay.size_of(&mut ops);
 		let overlay = self.journal_overlay.read();
 
-		mem += overlay.backing_overlay.size_of(&mut ops);
+		mem += overlay.cumulative_size; //This does not present real HashMap memory footprint but we are doing shrink_to_size after removal so this is good approximation.
```

Comment:
note that memory-db 0.24 includes https://github.com/paritytech/trie/pull/94, which should not be heavy to compute

Reply:
Hm, it looks like it is calculated same as cumulative_size. Thanks for the note! For now, I would like to just use comulative_size so as to not change much in code but this is something that we could use in the future.

Reply:
No, `cumulative_size` does not keep track the `HashMap` memory footprint as you noticed in the comments, while `overlay.backing_overlay.size_of(&mut ops)` does and it does it in O(1) so should not be heavy.

Reply:
Sorry, it seems I missed something. I will include the change, thanks again.